### PR TITLE
Feature/atr 271 add actions to graph semantic model

### PR DESCRIPTION
**Pull request by SENya1990**
Merge after ATR-118 and ATR-194

### CODE REVIEW

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/ActionHandlerInfo.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        public ActionHandlerInfo(MethodDeclarationSyntax node, IMethodSymbol symbol) : base(node, symbol)
+        {
+        }
+
+		[DebuggerBrowsable(DebuggerBrowsableState.Never)]
+		private string DebuggerDisplay => $"Symbol: {Symbol.Name}";
+
+		public ActionHandlerInfo(MethodDeclarationSyntax node, IMethodSymbol symbol, ActionHandlerInfo baseInfo) : this(node, symbol)
+        {
+            baseInfo.ThrowOnNull();
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/ActionHandlerInfo.cs]: nameof

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/DataViewInfo.cs [hunk 2]
```diff
@@ -1,41 +1,64 @@
...
+        public DataViewInfo(ISymbol symbol, INamedTypeSymbol type, PXContext pxContext, DataViewInfo baseInfo)
+            : this(symbol, type, pxContext)
         {
-            baseInfo.ThrowOnNull();
+            baseInfo.ThrowOnNull(nameof(baseInfo));
 
             Base = baseInfo;
         }
+
+        public void InitProcessingDelegatesInfo()
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/DataViewInfo.cs]: I forgot to delete this method, please delete it

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+			addGraphActionInfo(actionsByName, graphType);
+
+			if (baseType.TypeArguments.Length >= 2 && !AddInfoFromBaseExtensions())
+				return empty;
+
+			addGraphExtensionActionInfo(actionsByName, graphExtension);
+			return actionsByName.Items;
+
+
+			//-----------------------------------------------------------------Local Function----------------------------------------------
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs]: I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+																				CancellationToken cancellation)
+		{
+			graphExtension.ThrowOnNull(nameof(graphExtension));
+			actionsByName.ThrowOnNull(nameof(actionsByName));
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			return GetActionInfoFromGraphExtension<(MethodDeclarationSyntax, IMethodSymbol)>(
+				graphExtension, pxContext, AddHandlersFromGraph, AddHandlersFromGraphExtension);
+
+			//--------------------------------------------------------Local Functions----------------------------------------------------------
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs]: I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs [hunk 5]
```diff
@@ -10,100 +9,261 @@
...
+			var rawActionInfos = Type == GraphType.PXGraph
+				? Symbol.GetActionSymbolsWithTypesFromGraph(_pxContext)
+				: Symbol.GetActionsFromGraphExtensionAndBaseGraph(_pxContext);
+
+			return rawActionInfos.ToLookup(a => a.Item.ActionSymbol.Name, StringComparer.OrdinalIgnoreCase)
+								 .ToImmutableDictionary(group => group.Key,
+														group => CreateActionInfo(group.First()),
+														keyComparer: StringComparer.OrdinalIgnoreCase);
+
+			//--------------------------------------------------------Local Function--------------------------------------------
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs]: I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs [hunk 6]
```diff
@@ -10,100 +9,261 @@
...
+			var rawActionHandlerInfos = Type == GraphType.PXGraph
+				? Symbol.GetActionHandlersFromGraph(ActionsByNames, _pxContext, _cancellation)
+				: Symbol.GetActionHandlersFromGraphExtensionAndBaseGraph(ActionsByNames, _pxContext, _cancellation);
+
+			return rawActionHandlerInfos.ToLookup(handler => handler.Item.Symbol.Name, StringComparer.OrdinalIgnoreCase)
+										.ToImmutableDictionary(group => group.Key,
+															   group => CreateActionHandlerInfo(group.First()),
+															   keyComparer: StringComparer.OrdinalIgnoreCase);
+
+			//--------------------------------------------------------Local Function--------------------------------------------
```
Comment (metallmonster) [inline src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs]: I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/ActionHandlerInfo.cs (hunk 1 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/ActionHandlerInfo.cs]: ok

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/DataViewInfo.cs (hunk 2 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/DataViewInfo.cs]: ok

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs (hunk 3 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs]: It slightly helps me to distinguish local methods but I don't mind deleting them

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs (hunk 4 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs]: ok

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs (hunk 5 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs]: ok

File: src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs (hunk 6 above)
Comment (SENya1990) [reply src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs]: ok

**Comment by SENya1990** [comment]
@metallmonster @dropsonic  one more try

**Pull request merged**